### PR TITLE
[REF] Split eigendecomp into ICA and PCA files

### PR DESCRIPTION
**Pull request by tsalo**
Closes None. References #166. This was originally part of #247, but has been split off because it is self-contained and should be merged separately.

Changes proposed in this pull request:
- Split eigendecomp into ICA and PCA files.
- Move kundu_tedpca into selection module.
- Add out_dir argument to tedpca.
- Remove pcastate file. PCA doesn't take very long.
- Transpose PCA mixing matrix internally so it's in the same format as the ICA mixing matrix.

### CODE REVIEW

**Comment by emdupre** [comment]
Sorry, why did we want to put ICA and PCA in their own modules ?

**Comment by tsalo** [comment]
The other changes are perhaps more important (e.g., moving kundu_tedpca into the selection module and removing the pcastate file), but my thinking is that separating ICA and PCA will work better when we add new decomposition methods (one of our future goals). Plus, if we want to support low-memory incremental PCA, then that may merit an additional function within pca.py (like run_mlepca). If we end up expanding the ICA- and PCA-related functions, then it makes sense to group them within their own submodules.

File: tedana/decomposition/ica.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+"""
+Signal decomposition methods for tedana
```
Comment (emdupre) [inline tedana/decomposition/ica.py]:
```suggestion
ICA and related signal decomposition methods for tedana
```

File: tedana/decomposition/ica.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    maxit : :obj:`int`, optional
+        Maximum number of iterations for ICA. Default is 500.
+    maxrestart : :obj:`int`, optional
+        Maximum number of attempted decompositions to perform with different
+        random seeds. ICA will stop running if there is convergence prior to
+        reaching this limit. Default is 10.
+
+    Returns
+    -------
+    mmix : (T x C) :obj:`numpy.ndarray`
```
Comment (emdupre) [inline tedana/decomposition/ica.py]: Should the docstring include that this is z-scored ?

File: tedana/decomposition/pca.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
+"""
+Signal decomposition methods for tedana
```
Comment (emdupre) [inline tedana/decomposition/pca.py]:
```suggestion
PCA and related signal decomposition methods for tedana
```

File: tedana/decomposition/pca.py [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+      booktitle={Advances in neural information processing systems},
+      pages={598--604},
+      year={2001}
+    }
+    """),
+    description='Introduces method for choosing PCA dimensionality '
+                'automatically')
+def run_mlepca(data):
+    """
+    Run Singular Value Decomposition (SVD) on input data.
```
Comment (emdupre) [inline tedana/decomposition/pca.py]:
```suggestion
    Run Singular Value Decomposition (SVD) on input data,
    automatically select components on MLE variance cut-off.
```

File: tedana/decomposition/pca.py [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+        Dimensionality augmentation weight for Rho calculations. Must be a
+        non-negative float, or -1 (a special value). Default is 1.
+    out_dir : :obj:`str`, optional
+        Output directory.
+    verbose : :obj:`bool`, optional
+        Whether to output files from fitmodels_direct or not. Default: False
+
+    Returns
+    -------
+    n_components : :obj:`int`
```
Comment (emdupre) [inline tedana/decomposition/pca.py]: Can we flip these to match the output ordering ?

**Pull request merged**